tarodb/assets_store.go

```diff
@@ -660,6 +660,34 @@ func constraintsToDbFilter(query *AssetQueryFilters) QueryAssetFilters {
 	return assetFilter
 }
 
+// specificAssetFilter maps the given asset parameters to the set of filters
+// we use in the SQL queries.
+func specificAssetFilter(id asset.ID, anchorPoint wire.OutPoint, amt uint64,
+	groupKey *asset.GroupKey) (QueryAssetFilters, error) {
+
+	anchorPointBytes, err := encodeOutpoint(anchorPoint)
+	if err != nil {
+		return QueryAssetFilters{}, fmt.Errorf("unable to encode "+
+			"outpoint: %w", err)
+	}
+
+	filter := QueryAssetFilters{
+		AssetIDFilter: id[:],
+		AnchorPoint:   anchorPointBytes,
+		MinAmt: sql.NullInt64{
+			Int64: int64(amt),
+			Valid: true,
+		},
+	}
+
+	if groupKey != nil {
+		key := groupKey.GroupPubKey
+		filter.KeyGroupFilter = key.SerializeCompressed()
```

Comment:
isn't the asset ID enough to be able to uniquely pick out an asset?

Reply:
No, in case of a normal asset where I have multiple coins, we actually need the script key, which I added now. We don't necessarily need the group key for our specific query, but because this function creates a generic filter, I'm leaving that in for now.